types/wordpress__notices/package.json

```diff
@@ -0,0 +1,6 @@
+{
+    "private": true,
+    "dependencies": {
+        "@wordpress/data": "^8.5.0"
```

Comment:
Hey @marekdedic Could you help provide me with some context around this?

Reply:
Hi, of course. This PR is a part of my effort in #63638 to remove the `@types/wordpress__data` package as type information is now published as part of `@wordpress/data`. Only instead of making one giant PR that modifies 10 packages at once, I made 10 PRs, upgrading and fixing definitions along the way.

Reply:
Does this need to be a fixed version? This ends up requiring an outdated version of `@wordpress/data` in the root of my project.

```
@wordpress/data@8.6.0 dev
node_modules/@wordpress/data
  @wordpress/data@"^8.5.0" from @types/wordpress__notices@3.27.0
  node_modules/@types/wordpress__notices
    @types/wordpress__notices@"*" from @types/wordpress__components@23.0.1
    node_modules/@types/wordpress__components
      @types/wordpress__components@"*" from @types/wordpress__components@23.0.1
      dev @types/wordpress__components@"^23.0.1" from @myproject/welcome@0.1.0
      packages/welcome
        @myproject/welcome@0.1.0
        node_modules/@myproject/welcome
          workspace packages/welcome from the root project
```

Reply:
I think the DT policy is to have the dependency version constraints match those of the underlying package.

Reply:
Gotcha, that makes sense. But in this case the underlying package is not version constrained: https://github.com/WordPress/gutenberg/blob/trunk/packages/notices/package.json#L31

Reply:
It actually is - it just uses the local files to set the constraint. You can see the resulting contraint [here](https://www.npmjs.com/package/@wordpress/notices/v/3.27.0?activeTab=code).

Reply:
Isn't the local constraint is always the latest version of the package though? I'm probably misunderstanding something, sorry. This package ends up requiring an older version of the WP data library, even though the latest version of the library specifies the newest version.

Reply:
As far as I understand this, it is the latest version *at the time of that particular release* - when I'm figuring this out, I checkout the release tag for `@wordpress/notices` and navigate to the folder for `@wordpress/data` and read its version from its `package.json`